emscripten.py

```diff
@@ -1107,10 +1107,13 @@ def make_bad(target=None):
     end = raw.rindex(']')
     body = raw[start + 1:end].split(',')
     if shared.Settings.EMULATED_FUNCTION_POINTERS:
+      # this speeds up this section when all_implemented is very big > 150000
+      notinImplemented = list(set(body) - set(all_implemented))
```

Comment:
Wouldn't this be a lot faster if you dropped the list() since "if x in set" is O(1) rather than O(n) for "if x in list"?

Reply:
In fact we change `get_all_implemented` to return a set then I think that should improve performance here and elsewhere significantly.

As far as I can tell the ordering of `all_implemented` is not important anywhere.

Reply:
I took a stab a this, I hope you don't mind: https://github.com/emscripten-core/emscripten/pull/9435

Reply:
thanks @sbc100 let me test out your changes report the numbers

Reply:
thanks @sbc100 the reduction is similar. but your changes are better. Thanks! going to close this